include/net/lwm2m.h

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __LWM2M_H__
+#define __LWM2M_H__
+
+#include <net/net_context.h>
+
+/* LWM2M Objects defined by OMA */
+
+#define LWM2M_OBJECT_SECURITY_ID			0
+#define LWM2M_OBJECT_SERVER_ID				1
+#define LWM2M_OBJECT_ACCESS_CONTROL_ID			2
+#define LWM2M_OBJECT_DEVICE_ID				3
+#define LWM2M_OBJECT_CONNECTIVITY_MONITORING_ID		4
+#define LWM2M_OBJECT_FIRMWARE_ID			5
+#define LWM2M_OBJECT_LOCATION_ID			6
+#define LWM2M_OBJECT_CONNECTIVITY_STATISTICS_ID		7
+
+/* callback can return 1 if handled (don't update value) */
+typedef void *(*lwm2m_engine_get_data_cb_t)(u16_t obj_inst_id,
+				       size_t *data_len);
+typedef int (*lwm2m_engine_set_data_cb_t)(u16_t obj_inst_id,
+				       u8_t *data, u16_t data_len,
+				       bool last_block, size_t total_size);
+typedef int (*lwm2m_engine_exec_cb_t)(u16_t obj_inst_id);
+
+
+/* LWM2M Device Object */
+
+#define LWM2M_DEVICE_PWR_SRC_TYPE_DC_POWER	0
+#define LWM2M_DEVICE_PWR_SRC_TYPE_BAT_INT	1
+#define LWM2M_DEVICE_PWR_SRC_TYPE_BAT_EXT	2
+#define LWM2M_DEVICE_PWR_SRC_TYPE_UNUSED	3
+#define LWM2M_DEVICE_PWR_SRC_TYPE_PWR_OVER_ETH	4
+#define LWM2M_DEVICE_PWR_SRC_TYPE_USB		5
+#define LWM2M_DEVICE_PWR_SRC_TYPE_AC_POWER	6
+#define LWM2M_DEVICE_PWR_SRC_TYPE_SOLAR		7
+#define LWM2M_DEVICE_PWR_SRC_TYPE_MAX		8
+
+#define LWM2M_DEVICE_ERROR_NONE			0
+#define LWM2M_DEVICE_ERROR_LOW_POWER		1
+#define LWM2M_DEVICE_ERROR_EXT_POWER_SUPPLY_OFF	2
+#define LWM2M_DEVICE_ERROR_GPS_FAILURE		3
+#define LWM2M_DEVICE_ERROR_LOW_SIGNAL_STRENGTH	4
+#define LWM2M_DEVICE_ERROR_OUT_OF_MEMORY	5
+#define LWM2M_DEVICE_ERROR_SMS_FAILURE		6
+#define LWM2M_DEVICE_ERROR_NETWORK_FAILURE	7
+#define LWM2M_DEVICE_ERROR_PERIPHERAL_FAILURE	8
+
+#define LWM2M_DEVICE_BATTERY_STATUS_NORMAL	0
+#define LWM2M_DEVICE_BATTERY_STATUS_CHARGING	1
+#define LWM2M_DEVICE_BATTERY_STATUS_CHARGE_COMP	2
+#define LWM2M_DEVICE_BATTERY_STATUS_DAMAGED	3
+#define LWM2M_DEVICE_BATTERY_STATUS_LOW		4
+#define LWM2M_DEVICE_BATTERY_STATUS_NOT_INST	5
+#define LWM2M_DEVICE_BATTERY_STATUS_UNKNOWN	6
+
+int lwm2m_device_add_pwrsrc(u8_t pwr_src_type); /* returns index */
+int lwm2m_device_remove_pwrsrc(int index);
+int lwm2m_device_set_pwrsrc_voltage_mv(int index, int voltage_mv);
+int lwm2m_device_set_pwrsrc_current_ma(int index, int current_ma);
+int lwm2m_device_add_err(u8_t error_code);
+
+
+/* LWM2M Firemware Update Object */
+
+#define STATE_IDLE		0
+#define STATE_DOWNLOADING	1
+#define STATE_DOWNLOADED	2
+#define STATE_UPDATING		3
+
+#define	RESULT_DEFAULT		0
```

Comment:
This is a file I created, and I don't mind fixing spacing / readability issues like this.  Done.